### PR TITLE
Add native vector index support for neo4j lpg and fix vector filters

### PR DESCRIPTION
**Pull request by tomasonjo**
new neo4j supports vector indexes without prespecifying embedding dimension, which means we can use it in the LPG abstraction.

Also fixed vector filters

### CODE REVIEW

File: llama-index-integrations/graph_stores/llama-index-graph-stores-neo4j/llama_index/graph_stores/neo4j/neo4j_property_graph.py [hunk 1]
```diff
@@ -930,5 +951,29 @@ def get_schema_str(self, refresh: bool = False) -> str:
...
+            version_tuple = (*map(int, version.split("-")[0].split(".")), 0)
+        else:
+            version_tuple = tuple(map(int, version.split(".")))
+
+        target_version = (5, 23, 0)
+
+        if version_tuple >= target_version:
+            self._vector_index = True
+        else:
+            self._vector_index = False
```
Comment (logan-markewich) [inline llama-index-integrations/graph_stores/llama-index-graph-stores-neo4j/llama_index/graph_stores/neo4j/neo4j_property_graph.py:976]: Nit: the variable name feels a little misleading. I didn't realize this was a boolean until here. Maybe `self._supports_vector_index` is a better name?

**Review by logan-markewich — APPROVED** [review]
Lgtm -- we can also bump the version in the toml to publish this change, but maybe you wanted to wait for the next PR?

File: llama-index-integrations/graph_stores/llama-index-graph-stores-neo4j/llama_index/graph_stores/neo4j/neo4j_property_graph.py (hunk 1 above)
Comment (tomasonjo) [reply llama-index-integrations/graph_stores/llama-index-graph-stores-neo4j/llama_index/graph_stores/neo4j/neo4j_property_graph.py]: renamed

**Comment by tomasonjo** [comment]
Yeah, I'll add the fix as well tmr, should be fast, so let's wait with bumping the version

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**